app/forms/bulk_update/bulk_add_trainees_upload_form.rb

```diff
@@ -41,6 +38,26 @@ def csv
 
   private
 
+    def create_upload
+      upload = BulkUpdate::TraineeUpload.create(
+        provider: provider,
+        file: File.read(file),
+        file_name: file.original_filename,
+        number_of_trainees: csv&.count,
+        status: valid? ? :pending : :failed,
+      )
+
+      CSV.parse(upload.file, headers: :first_line).map.with_index do |row, index|
```

Comment:
Is there a `:first_line` option for the headers or it just evaluates to `true`?

```suggestion
      CSV.parse(upload.file, headers: true).map.with_index do |row, index|
```